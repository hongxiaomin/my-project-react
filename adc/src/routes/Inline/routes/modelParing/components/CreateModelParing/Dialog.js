import React from 'react'
import PropTypes from 'prop-types'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { injectIntl } from 'react-intl'
import { isEmpty } from 'lodash'
import { Button, Dialog } from '@blueprintjs/core'
import { Select, Checkbox } from 'antd'
import classes from './CreateModelParing.scss'
import { getModelOption, CHANGE_STATUS, MODEL_STATUS } from '../../modules/ModelParing'

const Option = Select.Option
export class DialogFilter extends React.Component {

  constructor(props) {
    super(props)
    this.state = {

    }
    this.handleChange = this.handleChange.bind(this)
    this.trim = this.trim.bind(this)
    this.onBlur = this.onBlur.bind(this)
    this.checkboxDefault = this.checkboxDefault.bind(this)
  }

  trim(str) {
    if (!str) return ''
    return str.replace(/(^\s*)|(\s*$)/g, '')
  }

  onBlur() {
    if (!isEmpty(this.props.that.state.processId) &&
     !isEmpty(this.props.that.state.productId)) {
      this.props.dispatch(getModelOption(this.props.that.state.processId,
         this.props.that.state.productId))
      this.props.dispatch({
        type: CHANGE_STATUS,
        payload: false,
      })
    }
    if (!isEmpty(this.props.that.state.processId) &&
     isEmpty(this.props.that.state.productId)) {
      this.props.dispatch(getModelOption(this.props.that.state.processId, ''))
      this.props.dispatch({
        type: CHANGE_STATUS,
        payload: false,
      })
    }
  }
  // 是否设置为试运行
  checkboxDefault(e) {
    this.props.that.setState({
      isTrialRun: e.target.checked,
    })
  }
  handleChange(type) {
    return (value) => {
      if (type === 'model') {
        this.props.dispatch({
          type: MODEL_STATUS,
          payload: value,
        })
      }
      this.props.that.setState({
        [type]: this.trim(value),
      })
    }
  }

  render() {
    const { intl } = this.props
    let defectCode
    if (!isEmpty(this.props.modelstatus) && !isEmpty(this.props.modelOption.modelList)) {
      defectCode = this.props.modelOption.modelList.filter((item) => {
        return item.modelId.toString() === this.props.that.state.model
      })
      defectCode = defectCode[0].defectList.join('/')
    }
    if (isEmpty(this.props.modelOption.modelList) && isEmpty(this.props.modelstatus)) {
      defectCode = []
    }
    return (
      <div>
        <Dialog
          isOpen={this.props.isOpen}
          onClose={this.props.toggleDialog}
          className={classes.dialog}
        >
          <div className={classes.header}>
            <h6 className={this.props.DialogID === '' ? classes.onlyHeaderTitle : classes.headerTitle}>
              {this.props.DialogTitle}
            </h6>
          </div>
          <div className={classes.dialogBody}>
            <div className={classes.filter}>
              <WrapperFormattedMessage id='processId' />
              <Select
                allowClear
                showSearch
                name='processId'
                optionFilterProp="children"
                style={{ width: 200 }}
                placeholder={intl.formatMessage({ id: 'pleaseSelect' })}
                onChange={this.handleChange('processId')}
                onBlur={this.onBlur}
              >
                {this.props.processId.map(item => (<Option
                  key={item.processName}
                  value={item.processId.toString()}
                >{item.processName}</Option>))}
              </Select>
            </div>
            <div className={classes.filter}>
              <WrapperFormattedMessage id='productId' />
              <Select
                allowClear
                showSearch
                optionFilterProp="children"
                name='productId'
                style={{ width: 200 }}
                placeholder={intl.formatMessage({ id: 'pleaseSelect' })}
                onChange={this.handleChange('productId')}
                onBlur={this.onBlur}
              >
                {this.props.productId.map(item => (<Option
                  key={item.productId}
                  value={item.productId.toString()}
                >{item.productName}</Option>))}
              </Select>
            </div>
            {
              !isEmpty(this.props.checkError) &&
              <div style={{ color: 'red', margin: '0 10px' }}>Tips：{this.props.checkError}</div>
            }
            <div className={classes.filter}>
              <WrapperFormattedMessage id='modelName' />
              <Select
                showSearch
                name='model'
                optionFilterProp="children"
                style={{ width: 200 }}
                placeholder={intl.formatMessage({ id: 'pleaseSelect' })}
                onChange={this.handleChange('model')}
              >
                {
                  !isEmpty(this.props.modelOption.modelList) ?
                  this.props.modelOption.modelList.map(item => (<Option
                    key={item.modelId}
                    value={item.modelId.toString()}
                  >{item.modelName}</Option>)) :
                  [].map(item => (<Option
                    key={item.modelId}
                    value={item.modelId.toString()}
                  >{item.modelName}</Option>))
                }
              </Select>
            </div>
            {
              this.props.status === true &&
              !isEmpty(this.props.modelstatus) ?
                <div className={classes.filterDefect}>
                  <WrapperFormattedMessage id='labelingTool.eda.defectCode' />
                  <div style={{ width: '200px',
                    wordWrap: 'break-word',
                    backgroundColor: '#eee',
                    margin: '10px',
                    borderRadius: '5px',
                    padding: '10px' }}
                  >
                    {defectCode}
                  </div>
                </div> :
                <div />
            }
            <div className={classes.filter}>
              <WrapperFormattedMessage id='modelPairing.dialogFilter.trialRun' />
              <Checkbox
                style={{ width: '200px', marginLeft: '50px' }}
                checked={this.props.that.state.isTrialRun}
                onChange={this.checkboxDefault}
              ><WrapperFormattedMessage id='modelPairing.dialogFilter.setAsTrialRun' /></Checkbox>
            </div>
            <div className={classes.btnGroup}>
              <Button className={classes.btnCancel} onClick={this.props.toggleDialog}>
                <WrapperFormattedMessage id='cancel' />
              </Button>
              <Button
                className={`pt-button pt-intent-primary ${classes.btnEdit}`}
                onClick={this.props.handleSubmit}
              >
                <WrapperFormattedMessage id='Save' />
              </Button>
            </div>
          </div>
        </Dialog>
      </div>
    )
  }
}

DialogFilter.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  DialogID: PropTypes.string.isRequired,
  DialogTitle: PropTypes.string.isRequired,
  processId: PropTypes.array.isRequired,
  productId: PropTypes.array.isRequired,
  modelOption: PropTypes.object.isRequired,
  modelstatus: PropTypes.string.isRequired,
  status: PropTypes.bool.isRequired,
  toggleDialog: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  that: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
  checkError: PropTypes.string.isRequired,
  intl: PropTypes.object.isRequired,
}

export default injectIntl(DialogFilter)
