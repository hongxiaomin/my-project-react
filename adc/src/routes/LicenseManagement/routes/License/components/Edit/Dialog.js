import React from 'react'
import PropTypes from 'prop-types'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { injectIntl } from 'react-intl'
import { Form, Input, Select, Icon, Button, Modal } from 'antd'
import { trim } from 'lodash'
import { getRole, showMessage } from 'Utils'

const Option = Select.Option
const FormItem = Form.Item
export class DialogFilter extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
      loading: false,
      passwordInputType: 'password',
    }
    this.handleChange = this.handleChange.bind(this)
  }

  mouseDownEye = () => {
    console.log('this.mouseDown')
    this.setState({
      passwordInputType: 'text',
    })
  }

  mouseUpEye = () => {
    this.setState({
      passwordInputType: 'password',
    })
  }

  handleChange(e) {
    e.preventDefault()
    this.props.form.validateFields((err, objValues) => {
      if (!err) {
        this.setState({
          loading: true,
        })
        const postData = {}
        const keys = Object.keys(objValues)
        const values = Object.values(objValues)
        values.forEach((value, index) => (postData[keys[index]] = trim(value)))
        postData.roleIds = [postData.roleIds]

        this.props.handleSubmit(postData, () => {
          this.setState({
            loading: false,
          })
          this.props.toggleDialog()
          this.props.form.setFieldsValue({
            userId: '',
            userName: '',
            password: '',
            roleIds: '',
          })
          // 选中的数据清空
        }, () => {
          this.setState({
            loading: false,
          })
          showMessage({ text: 'Operation failed!', type: 'error' })
        })
      }
    })
  }

  render() {
    const { intl } = this.props
    const { getFieldDecorator } = this.props.form
    const formItemLayout = {
      labelCol: {
        xs: { span: 24 },
        sm: { span: 6 },
      },
      wrapperCol: {
        xs: { span: 24 },
        sm: { span: 18 },
      },
    }
    return (
      <Modal
        visible={this.props.isOpen}
        title={this.props.dialogTitle}
        onCancel={this.props.toggleDialog}
        footer={[
          <Button key="cancel" onClick={this.props.toggleDialog}>Cancel</Button>,
          <Button key="submit" type="primary" loading={this.state.loading} onClick={this.handleChange}>
            <WrapperFormattedMessage id='Save' />
          </Button>,
        ]}
      >
        <div style={{ padding: '0 20px 0 20px' }}>
          <Form>
            <FormItem
              {...formItemLayout}
              label={intl.formatMessage({ id: 'userAndLicenseManagement.license.userID' })}
            >
              {getFieldDecorator('userId', {
                rules: [{ required: true, message: intl.formatMessage({ id: 'userAndLicenseManagement.edit.userIDError' }) }],
              })(
                <Input disabled={this.props.dialogBtnContent.toLowerCase() === 'update'} type="text" placeholder={intl.formatMessage({ id: 'userAndLicenseManagement.license.userID' })} />,
              )}
            </FormItem>
            <FormItem
              {...formItemLayout}
              label={intl.formatMessage({ id: 'userAndLicenseManagement.license.userName' })}
            >
              {getFieldDecorator('userName', {
                rules: [{ required: true, message: intl.formatMessage({ id: 'userAndLicenseManagement.edit.userNameError' }) }],
              })(
                <Input type="text" placeholder="User Name" />,
              )}
            </FormItem>
            <FormItem
              {...formItemLayout}
              label={this.props.dialogBtnContent !== 'Update' ? 'Password' : <span style={{ paddingLeft: '10px' }}><WrapperFormattedMessage id='password' /></span>}
            >
              {getFieldDecorator('password', {
                rules: [{ required: this.props.dialogBtnContent !== 'Update', message: intl.formatMessage({ id: 'userAndLicenseManagement.edit.passwordError' }) }],
              })(
                <Input
                  suffix={<Icon
                    type="eye"
                    onMouseDown={this.mouseDownEye}
                    onMouseUp={this.mouseUpEye}
                  />}
                  type={this.state.passwordInputType}
                  placeholder={intl.formatMessage({ id: 'password' })}
                />,
              )}
            </FormItem>
            <FormItem
              {...formItemLayout}
              label={intl.formatMessage({ id: 'role' })}
            >
              {getFieldDecorator('roleIds', {
                rules: [{ required: true, message: intl.formatMessage({ id: 'userAndLicenseManagement.edit.roleError' }) }],
              })(
                <Select >
                  {(() => {
                    const roles = getRole()
                    const keys = Object.keys(roles)
                    const values = Object.values(roles)
                    return values.map((name, i) => (<Option
                      key={keys[i]}
                      value={keys[i]}
                    ><WrapperFormattedMessage id={name} /></Option>))
                  })()}
                </Select>
              )}
            </FormItem>

          </Form>
        </div>
      </Modal>
    )
  }
}


DialogFilter.propTypes = {
  dialogTitle: PropTypes.string.isRequired,
  dialogBtnContent: PropTypes.string.isRequired,
  isOpen: PropTypes.bool.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  toggleDialog: PropTypes.func.isRequired,
  form: PropTypes.object.isRequired,
}

export default Form.create()(injectIntl(DialogFilter))
