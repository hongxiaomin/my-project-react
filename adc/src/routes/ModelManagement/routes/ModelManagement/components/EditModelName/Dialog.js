import React from 'react'
import { assign } from 'lodash'
import classes from './EditModelName.scss'

type Props = {
  that: Object,
};
class Dialog extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
    }
  }

  inputChange(type) {
    return (e) => {
      this.props.that.setState(assign({}, this.props.that.state, {
        [type]: e.target.value,
      }))
    }
  }

  props: Props;
  render() {
    return (
      <div className={classes['OneStep-container']}>
        <div className={classes.dialogBody}>
          <div>
            <div className={classes.filter}>
              Model Name:
              <input
                type='text'
                className={'pt-input'}
                value={this.props.that.state.modelName}
                onChange={this.inputChange('modelName')}
              />
            </div>
            <div className={classes.filter}>
              Description:
              <textarea
                style={{ width: '200px' }}
                className={'pt-input'}
                value={this.props.that.state.description}
                onChange={this.inputChange('description')}
              />
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default Dialog
