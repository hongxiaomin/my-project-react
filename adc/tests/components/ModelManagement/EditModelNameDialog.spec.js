import React from 'react'
import { shallow } from 'enzyme'
import Dialog from 'routes/ModelManagement/routes/ModelManagement/components/EditModelName/Dialog'

describe('(Component) Dialog', () => {
  const that = { state: { modelName: 'model' } }
  const wrapper = shallow(
    <Dialog
      that={that}
    />,
  )
  it('render normally', () => {
    expect(wrapper).toMatchSnapshot()
  })
})