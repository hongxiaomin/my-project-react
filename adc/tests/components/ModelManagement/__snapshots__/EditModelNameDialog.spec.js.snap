// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Dialog render normally 1`] = `
<div
  className="OneStep-container"
>
  <div
    className="dialogBody"
  >
    <div>
      <div
        className="filter"
      >
        Model Name:
        <input
          className="pt-input"
          onChange={[Function]}
          type="text"
          value="model"
        />
      </div>
      <div
        className="filter"
      >
        Description:
        <textarea
          className="pt-input"
          onChange={[Function]}
          style={
            Object {
              "width": "200px",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
